src/Raven.Server/Documents/Sharding/Subscriptions/OrchestratedSubscriptionConnection.cs

```diff
@@ -130,6 +131,24 @@ protected override void AssertSupportedFeatures()
                 throw new NotSupportedInShardingException("Concurrent subscriptions are not supported in sharding.");
         }
 
+        protected override void AssertCloseWhenNoDocsLeft()
+        {
+            if (_options.CloseWhenNoDocsLeft)
```

Comment:
instead of this... how about to set an exception to `_waitForMoreDocuments` if all shards are closed?

Reply:
I will keep the exception, its more straight forward